plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts

```diff
@@ -116,4 +130,208 @@ describe('KubernetesProxy', () => {
+    worker.use(
+      rest.get('https://localhost:9999/api/v1/namespaces', (req, res, ctx) => {
+        if (!req.headers.get('Authorization')) {
+          return res(ctx.status(401));
+        }
+
+        if (req.headers.get('Authorization') !== 'Bearer randomToken') {
+          return res(ctx.status(403));
+        }
+
+        return res(
+          ctx.status(200),
+          ctx.json({
+            kind: 'NamespaceList',
+            apiVersion: 'v1',
+            items: [],
+          }),
+        );
+      }),
+    );
+
+    clusterSupplier.getClusters.mockResolvedValue([
+      {
+        name: 'cluster1',
+        url: 'https://localhost:9999',
+        serviceAccountToken: 'randomToken',
+        authProvider: 'serviceAccount',
+      },
+    ] as ClusterDetails[]);
+
+    authTranslator.decorateClusterDetailsWithAuth.mockResolvedValue({
+      name: 'cluster1',
+      url: 'https://localhost:9999',
+      serviceAccountToken: 'randomToken',
+      authProvider: 'serviceAccount',
+    } as ClusterDetails);
+
+    const app = express().use('/mountpath', proxy.createRequestHandler());
+    const requestPromise = request(app)
+      .get('/mountpath/api/v1/namespaces')
+      .set(HEADER_KUBERNETES_CLUSTER, 'cluster1');
+
+    worker.use(rest.all(requestPromise.url, req => req.passthrough()));
+
+    const response = await requestPromise;
+
+    expect(response.status).toEqual(200);
+  });
+
+  it('should add the serviceAccountToken provided by cluster details', async () => {
+    worker.use(
+      rest.get('https://localhost:9999/api/v1/namespaces', (req, res, ctx) => {
+        if (!req.headers.get('Authorization')) {
+          return res(ctx.status(401));
+        }
+
+        if (req.headers.get('Authorization') !== 'Bearer catToken') {
+          return res(ctx.status(403));
+        }
+
+        return res(
+          ctx.status(200),
+          ctx.json({
+            kind: 'NamespaceList',
+            apiVersion: 'v1',
+            items: [],
+          }),
+        );
+      }),
+    );
+
+    clusterSupplier.getClusters.mockResolvedValue([
+      {
+        name: 'cluster1',
+        url: 'https://localhost:9999',
+        serviceAccountToken: 'catToken',
+        authProvider: 'serviceAccount',
+      },
+    ] as ClusterDetails[]);
+
+    authTranslator.decorateClusterDetailsWithAuth.mockResolvedValue({
+      name: 'cluster1',
+      url: 'https://localhost:9999',
+      serviceAccountToken: 'catToken',
+      authProvider: 'serviceAccount',
+    } as ClusterDetails);
+
+    const app = express().use('/mountpath', proxy.createRequestHandler());
+    const requestPromise = request(app)
+      .get('/mountpath/api/v1/namespaces')
+      .set(HEADER_KUBERNETES_CLUSTER, 'cluster1');
+
+    worker.use(rest.all(requestPromise.url, req => req.passthrough()));
+
+    const response = await requestPromise;
+
+    expect(response.status).toEqual(200);
+  });
```

Comment:
I'm pretty sure this test is redundant to the `should add a serviceAccountToken to the request headers if one isnt provided in request and one isnt set up in cluster details` case below -- this test talks about a token being present in the clusterDetails, but then the translator is mocked to return a token anyway. Effectively this is simulating the case where the translator getting invoked by the `KubernetesAuthTranslatorGenerator` is the `NoopKubernetesAuthTranslator`.
```suggestion
```

Reply:
Redundant test was removed, i think i was trying to cover the case where we have cluster details with an `serviceAccountToken` already and make it so we would just default to using those credentials rather than rely on the auth translator. But after working out the feature the current implementation makes it so we always make use of an authTranslator regardless of the content of clusterDetails which was done by design. Therefore, I agree that i can remove this redundant test and keep the `should add a serviceAccountToken to the request headers if one isn't provided in request and one isn't set up in cluster details` test which is what I was trying to accomplish with this feature. Done.